ocaml/util/xapi_version.ml

```diff
@@ -9,5 +9,9 @@
 let git_id = ""
 let hostname = "localhost"
 let date = Build_info.date
-let xapi_version_major = 1
-let xapi_version_minor = 20
+let (xapi_version_major, xapi_version_minor) =
+  try
+    Scanf.sscanf Build_info.version "%d.%d.%s" (fun maj min _rest -> (maj, min))
+  with _ ->
+    Printf.eprintf "Could not determine xapi_version (are you building with opam?) defaulting to (0,0)";
+    (0, 0)
```

Comment:
As discussed it would be good to make this fail.
There is already a .gitarchive-info in this folder, but that only tells you the commit hash, and since opam just creates something like 'master.tar.gz' (for xs-opam), the actual version is lost.

Would a suitable workaround be to set in the opam file: `XAPI_VERSION=v0.0.0` in the build command? And then we can make this fail.